src/Illuminate/Container/Container.php

```diff
@@ -678,6 +678,31 @@ public function makeWith($abstract, array $parameters = [])
         return $this->make($abstract, $parameters);
     }
 
+    /**
+     * Resolve the given type from the container or execute the callback.
+     *
+     * @param  string  $abstract
+     * @param  mixed|array  $parameters
+     * @param  \Closure|null  $callback
+     * @return mixed
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
+     */
+    public function makeOr($abstract, $parameters = [], Closure $callback = null)
+    {
+        try {
+            if ($parameters instanceof Closure) {
+                $callback = $parameters;
```

Comment:
what does this do?

Reply:
@bert-w It allows you to pass the callback as a second argument preserving the method signature and avoiding parsing the args list.

Reply:
Oh I see now; this can fail and the catch-clause could be called